src/main/scala/beam/utils/OptionalUtils.scala

```diff
@@ -30,4 +32,8 @@ object OptionalUtils {
       */
     def toOption: Option[T] = if (opt.isPresent) Some(opt.get()) else None
   }
+
+  implicit class OptionalTimeExtension(optionalTime: OptionalTime) {
+    def toOption: Option[Double] = if (optionalTime.isDefined) Option(optionalTime.seconds()) else Option.empty[Double]
```

Comment:
We could opt to make this return a Some(Double.NegativeInfinity) for backwards compat.